src/term.rs

```diff
@@ -417,6 +418,73 @@ impl ArrayAttrs {
     }
 }
 
+pub mod record {
+    use super::{RecordAttrs, RichTerm};
+    use crate::identifier::Ident;
+    use std::collections::HashMap;
+
+    /// The base structure of a Nickel record.
+    ///
+    /// Used to group together fields common to both the `Record` and
+    /// `RecRecord` terms.
+    #[derive(Clone, Debug, PartialEq)]
+    pub struct Record {
+        /// Fields whose names are known statically.
+        pub fields: HashMap<Ident, RichTerm>,
+        /// Attributes which may be applied to a record.
+        pub attrs: RecordAttrs,
+    }
+
+    impl Record {
+        pub fn new(fields: HashMap<Ident, RichTerm>, attrs: RecordAttrs) -> Self {
+            Record { fields, attrs }
+        }
+
+        /// A record with no fields and the default set of attributes.
+        pub fn empty() -> Self {
+            let fields = HashMap::new();
+            let attrs = Default::default();
+            Record { fields, attrs }
+        }
```

Comment:
Rather than having an ad hoc function, can we/should we have `Record` implement the [Default](https://doc.rust-lang.org/std/default/trait.Default.html) trait?

Reply:
It is customary to have a default implementation indeed. You can probably just derive it for `Record` using `#[derive(Default)]`, because the subpart implements `Default`, I think. Then it's standard to add a `new()` method which is just an alias for `default()`.